bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/NfcResource.java

```diff
@@ -188,27 +224,67 @@ public Response deprecatedGetStore(
         return response;
     }
 
+    @Path( "/{packageType}/{type: (hosted|group|remote)}/{name}/info" )
+    @ApiOperation( "Get not-found cache information, e.g., size, etc" )
+    @ApiResponses(
+                    { @ApiResponse( code = 200, response = NotFoundCacheInfoDTO.class, message = "The info of not-found cache" ) } )
+    @GET
+    @Produces( ApplicationContent.application_json )
+    public Response getStoreInfo(
+                    final @ApiParam( name = "packageType", required = true, value = "type of package (eg. maven, npm, generic-http)" )
+                    @PathParam( "packageType" ) String packageType,
+                    final @ApiParam( allowableValues = "hosted,group,remote", name = "type", required = true, value = "type of store" )
+                    @PathParam( "type" ) String t,
+                    final @ApiParam( name = "name", value = "name of the store" )
+                    @PathParam( "name" ) String name )
+    {
+        Response response;
+        final StoreType type = StoreType.get( t );
+        final StoreKey key = new StoreKey( packageType, type, name );
+        try
+        {
+            NotFoundCacheInfoDTO dto = controller.getInfo( key );
+            response = formatOkResponseWithJsonEntity( dto, serializer );
+        }
+        catch ( final IndyWorkflowException e )
+        {
+            response = formatResponse( e );
+        }
+        return response;
+    }
+
     @Path( "/{packageType}/{type: (hosted|group|remote)}/{name}" )
```

Comment:
Params: pageIndex and pageSize should also be added in the @ Path here. same for methods: deprecatedGetStore, getAll

Reply:
I use @QueryParam instead of @PathParam for pagination. For such optional args using queryParam can keep the path patterns simple.